sdk/daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/UpgradeTest.scala

```diff
@@ -936,6 +937,230 @@ class UpgradeTest extends AnyFreeSpec with Matchers {
       s"""throw @'$commonDefsPkgId':Mod:MyView @'$commonDefsPkgId':Mod:Ex ('$commonDefsPkgId':Mod:Ex {message = "View"})"""
   }
 
+  case object ValidDowngradeAdditionalTemplateArg
+      extends TestCase("ValidDowngradeAdditionalTemplateArg", ExpectSuccess) {
+    override def v1AdditionalFields = ", extra: Option Unit"
```

Comment:
Why do we set the extra field on v1 instead of v2? I assume it's because we don't actually use the 1.0.0/2.0.0 versions anywhere, so v1/v2 only determines the order in which we run our exercises?

If so, would it be useful to have a more descriptive name than v1/v2? I've opted for first/second.

Reply:
Yes exactly, the engine doesn't care about version numbers. I agree that the names are misleading but I think I want to have a way to swap them instead so I'll address that in a further PR.